### PR TITLE
Add overscroll haptics.

### PR DESCRIPTION
**Pull request by Kpeved**
#### WHAT
Add overscroll haptics

#### WHY
There were no overscroll haptics. Also by reaching the edge of the list haptics continued to fire. Now it called only once and no subsequent haptics happen.

#### HOW
A new interface RotaryHapticHandler was added, which will be used for handling haptics for scroll and snap scenarios. 
.rotary modifiers (rotaryWithSnap, rotaryWithScroll and rotaryWithFling) now accept RotaryHapticHandler as a parameter instead of RotaryHapticFeedback

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt [hunk 1]
```diff
@@ -108,7 +108,7 @@ public fun Modifier.rotaryWithFling(
     focusRequester: FocusRequester,
     scrollableState: ScrollableState,
     flingBehavior: FlingBehavior = ScrollableDefaults.flingBehavior(),
-    rotaryHaptics: RotaryHapticFeedback = rememberRotaryHapticFeedback(),
+    rotaryHaptics: RotaryHapticHandler = rememberRotaryHapticHandler(scrollableState),
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt:111]: You'll need to update the api files with `./gradlew metalavaGenerateSignature metalavaGenerateSignatureDebug`

**Comment by yschimke** [comment]
The API changes will break binary compatibility, but it seems unlikely people actively use the changed param, so change looks like a solid improvement.

**Review by yschimke — APPROVED** [review]
Pending the api updates.

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 2]
```diff
@@ -65,6 +67,79 @@ internal fun <T> Flow<T>.throttleLatest(timeframe: Long): Flow<T> =
...
+ * on the [scrollableState], scrolled pixels and [hapticsThreshold].
+ * Haptic is not fired in this class, instead it's sent to [hapticsChannel]
+ * where it'll performed later.
+ *
+ * @param scrollableState Haptic performed based on this state
+ * @param hapticsChannel Channel to which haptic events will be sent
+ * @param hapticsThreshold A scroll threshold after which haptic is produced.
+ */
+@OptIn(ExperimentalHorologistApi::class)
+public class DefaultRotaryHapticHandler(
```
[the remark this answers is not in the document]
Comment (Kpeved) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:100]: This is implementation of the interface. We might change this implementation in the future or remove interface altogether

**Pull request merged**